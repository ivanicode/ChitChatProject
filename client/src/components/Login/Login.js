import React from 'react';
import { useLoginHooks } from './loginHooks';
import { func } from 'prop-types';

export function Login({setUserData}){

    const {
        submitLogin,
        onLoginChangeHandler,
        loginData,
        saveState
    } = useLoginHooks(setUserData)

    const errorMessage = saveState.error ? (<div className="errorTextForNoUser">
        Nie ma takiego użytkownika
    </div>) : null;

    return (
        <div>
            <form onSubmit={submitLogin} className="loginInputForm">
                <div className="logo">
                    <img src="/logo.png" alt="logo"></img>
                </div>
                {errorMessage}
                <div className="login">
                    <div className="loginLabels">
                        <label className="loginInput">
                            Login:
                            <input type="text" id="login" value={loginData.login} onChange={onLoginChangeHandler} required></input>
                        </label>
                        <label className="loginPasswordInput">
                            Hasło:
                            <input type="password" id="loginPassword" value={loginData.loginPassword} onChange={onLoginChangeHandler} required></input>
                        </label>
                    </div>
                    <div className="loginButtonDiv">
                        <button className="loginButton" type="submit" onClick={submitLogin}>
                            Zaloguj
                        </button>
                    </div>
                </div>
            </form>
        </div>
    )
}

Login.displayName = 'Login';
Login.propTypes = {
    setUserData: func
}
export default Login;