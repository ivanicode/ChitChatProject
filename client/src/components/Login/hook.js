import {useState, useEffect, useReducer} from 'react';
import dayjs from 'dayjs';


export function useLoginHooks(){
    return {submitLogin}
}

const {saveData} = useSave('/Login/api/user')

function submitLogin(event){
    saveData()
}

export function useSave(path) {
    const [saveState, dispatch] = useReducer(reducer, initialData);
  
    function saveData(data){
        dispatch({ type: 'requesting' });
        fetch(path, {
            method: 'POST',
            headers: {
                'Content-Type': 'application/json'
            },
            body: JSON.stringify(data)
        })
        .then(() => {
            dispatch({ type: 'success'});
        })
        .catch(error => {
            dispatch({ type: 'error', error});
            console.error(error);
        })
    }
  
    return {saveState, saveData};
} 