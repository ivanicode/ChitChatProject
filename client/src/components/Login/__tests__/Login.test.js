import React from 'react';
import {shallow} from 'enzyme';
import {Login} from '../Login';
import * as hook from '../hook';

describe('Login component', () => {
    it('should match snapshot when no errors', () => {
        hook.useLoginHooks = jest.fn().mockImplementation(
            () => ({
                loginData: {},
                saveState: {},
                submitLogin: jest.fn(),
                onLoginChangeHandler: jest.fn()
            })
        )
        const wrapper = shallow(<Login />)
        expect(wrapper).toMatchSnapshot()
    })
    it('should match snapshot when error exist', () => {
        hook.useLoginHooks = jest.fn().mockImplementation(
            () => ({
                loginData: {},
                saveState: {error: {}},
                submitLogin: jest.fn(),
                onLoginChangeHandler: jest.fn()
            })
        )
        const wrapper = shallow(<Login />)
        expect(wrapper).toMatchSnapshot()
    } )
})