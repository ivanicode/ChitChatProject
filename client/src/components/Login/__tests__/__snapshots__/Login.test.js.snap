// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Login component should match snapshot when error exist 1`] = `
<div>
  <form
    className="loginInputForm"
    onSubmit={[MockFunction]}
  >
    <div
      className="logo"
    >
      <img
        alt="logo"
        src="/logo.png"
      />
    </div>
    <div
      className="errorTextForNoUser"
    >
      Nie ma takiego użytkownika
    </div>
    <div
      className="login"
    >
      <div
        className="loginLabels"
      >
        <label
          className="loginInput"
        >
          Login:
          <input
            id="login"
            onChange={[MockFunction]}
            required={true}
            type="text"
          />
        </label>
        <label
          className="loginPasswordInput"
        >
          Hasło:
          <input
            id="loginPassword"
            onChange={[MockFunction]}
            required={true}
            type="password"
          />
        </label>
      </div>
      <div
        className="loginButtonDiv"
      >
        <button
          className="loginButton"
          onClick={[MockFunction]}
          type="submit"
        >
          Zaloguj
        </button>
      </div>
    </div>
  </form>
</div>
`;

exports[`Login component should match snapshot when no errors 1`] = `
<div>
  <form
    className="loginInputForm"
    onSubmit={[MockFunction]}
  >
    <div
      className="logo"
    >
      <img
        alt="logo"
        src="/logo.png"
      />
    </div>
    <div
      className="login"
    >
      <div
        className="loginLabels"
      >
        <label
          className="loginInput"
        >
          Login:
          <input
            id="login"
            onChange={[MockFunction]}
            required={true}
            type="text"
          />
        </label>
        <label
          className="loginPasswordInput"
        >
          Hasło:
          <input
            id="loginPassword"
            onChange={[MockFunction]}
            required={true}
            type="password"
          />
        </label>
      </div>
      <div
        className="loginButtonDiv"
      >
        <button
          className="loginButton"
          onClick={[MockFunction]}
          type="submit"
        >
          Zaloguj
        </button>
      </div>
    </div>
  </form>
</div>
`;
